src/algebra/module/linear_map.lean

```diff
@@ -44,38 +61,45 @@ section
 
 set_option old_structure_cmd true
 
-/-- A map `f` between modules over a semiring is linear if it satisfies the two properties
-`f (x + y) = f x + f y` and `f (c • x) = c • f x`. Elements of `linear_map R M M₂` (available under
-the notation `M →ₗ[R] M₂`) are bundled versions of such maps. An unbundled version is available with
-the predicate `is_linear_map`, but it should be avoided most of the time. -/
-structure linear_map (R : Type u) (M : Type v) (M₂ : Type w)
-  [semiring R] [add_comm_monoid M] [add_comm_monoid M₂] [module R M] [module R M₂]
-  extends add_hom M M₂, M →[R] M₂
+/-- A map `f` between an `R`-module and an `S`-module over a ring homomorphism `σ : R →+* S`
+is semilinear if it satisfies the two properties `f (x + y) = f x + f y` and
+`f (c • x) = (σ c) • f x`. Elements of `linear_map σ M M₂` (available under the notation
+`M →ₛₗ[σ] M₂`) are bundled versions of such maps. For plain linear maps (i.e. for which
+`σ = ring_hom.id R`, the notation `M →ₗ[R] M₂` is available. An unbundled version of plain linear
+maps is available with the predicate `is_linear_map`, but it should be avoided most of the time. -/
+structure linear_map {R : Type*} {S : Type*} [semiring R] [semiring S]  (σ : R →+* S)
+  (M : Type*) (M₂ : Type*)
+  [add_comm_monoid M] [add_comm_monoid M₂] [module R M] [module S M₂]
+  extends add_hom M M₂ :=
+(map_smul' : ∀ (r : R) (x : M), to_fun (r • x) = (σ r) • to_fun x)
 
 end
 
 /-- The `add_hom` underlying a `linear_map`. -/
 add_decl_doc linear_map.to_add_hom
 
-/-- The `mul_action_hom` underlying a `linear_map`. -/
-add_decl_doc linear_map.to_mul_action_hom
-
-notation M ` →ₗ[`:25 R:25 `] `:0 M₂:0 := linear_map R M M₂
+notation M ` →ₛₗ[`:25 σ:25 `] `:0 M₂:0 := linear_map σ M M₂
+notation M ` →ₗ[`:25 R:25 `] `:0 M₂:0 := linear_map (ring_hom.id R) M M₂
 
 namespace linear_map
 
 section add_comm_monoid
 
-variables [semiring R] [add_comm_monoid M] [add_comm_monoid M₂] [add_comm_monoid M₃]
+variables [semiring R] [semiring S]
 
 section
-variables [module R M] [module R M₂]
+variables [add_comm_monoid M] [add_comm_monoid M₁] [add_comm_monoid M₂] [add_comm_monoid M₃]
+variables [add_comm_monoid N₁] [add_comm_monoid N₂] [add_comm_monoid N₃]
+variables [module R M] [module R M₂] [module S M₃]
+
+/-- The `mul_action_hom` underlying a `linear_map`. -/
+def to_mul_action_hom (f : M →ₗ[R] M₂) : mul_action_hom R M M₂ := {..f}
```

Comment:
This isn't worth having any more, it's already accessible via `f.to_distrib_mul_action_hom.to_mul_action_hom`, and having multiple ways to spell the same thing is usually not a good idea.